frontend/tests/services/withFeatureFlag.test.tsx

```diff
@@ -22,28 +21,41 @@ jest.mock("next/headers", () => ({
   cookies: () => Cookies,
 }));
 
+const searchPromise = (query: { [key: string]: string }) =>
+  new Promise<{ [key: string]: string }>((resolve) => {
+    resolve(query);
+  });
+
 describe("WithFeatureFlag", () => {
   afterEach(() => {
     enableFeature = false;
   });
   afterAll(() => {
     jest.restoreAllMocks();
   });
-  it("adds search params as prop to wrapped component", () => {
+  it("adds search params as prop to wrapped component", async () => {
     const OriginalComponent = jest.fn();
     const searchParams = { any: "param" };
     const WrappedComponent = withFeatureFlag(
       OriginalComponent,
       "searchOff",
       identity,
     );
-    render(<WrappedComponent searchParams={searchParams} />);
```

Comment:
This is a little gnarly. The FF wrapper is now async since it uses cookies and takes params.

Reply:
Yeah, this component in general is pretty gnarly. I'd welcome just ignoring typescript on this file if that'd make things easier. Maybe just make it a js file, or do we have a rule against that?